src/profession.cpp

```diff
@@ -428,6 +428,54 @@ std::list<item> profession::items( bool male, const std::vector<trait_id> &trait
         }
     }
 
+    /*  Purpose:    Post processing on profession selection and generation on start.
+                    Professions are newly generated each time on selection, even
+                    if the profession has already been selected previously. The profession
+                    is even generated again after selection, and the game is started.
+                    The purpose of this loop is to provide default numbers for item charges
+                    and ammo. This extends to food as well, however due to the nature of
+                    the way these professions are generated, food items one see's on the profession
+                    screen will be different in-game, but still these charge numbers will be
+                    dictated by the default of said items.
+
+        Graphical Issue:    Currently, worn items can fluctuate between 0 and their default value
+                            upon entering the game this issue will resolve itself and the correct
+                            default value will be given to the player. This issue occurs
+                            on holding the enter key to make rapid selection of a profession.
+
+        TODO:   Currently the way magazines are implemented they do not contain a default value.
+                Professions contain a max charge value found in the JSON file however, this
+                does not help in choosing a default. Below is a compromise that takes the
+                half of the magazines capacity as a default. This was chosen because most
+                defaults are half of the items max value.
+       -- Ideally, the default value of a magazine would be defined in the profession JSON file. --
+    */
```

Comment:
👏 👏 👏 

Always happy to seed thorough documentation. Well done on this feature, and good on you for writing stellar docs! 🎉